gammapy/datasets/fermi.py

```diff
@@ -216,6 +218,135 @@ def fetch_fermi_diffuse_background_model(filename='gll_iem_v02.fit'):
 
     return filename
 
+class Fermi3FGLObject(object):
+    """TODO: Doccomments
+    """
+    fermi_cat = fetch_fermi_catalog('3FGL')
```

Comment:
Putting the `fetch_fermi_catalog('3FGL')` call as a statement in class scope means it's executed on import.
This leads to this error and is generally surprising behaviour (downloading a file on import):
https://travis-ci.org/gammapy/gammapy/jobs/72960225#L1191

I'm not sure what the best solution here is. This seems to work:
```python
In [1]: %paste
class Fermi3FGLObject(object):
    
    # Fermi catalog is lazily loaded on first access
    # and cached at class level (not instance level)
    _fermi_cat = None

    def __init__(self):
        cat = self.fermi_cat()
    
    @classmethod
    def fermi_cat(cls):
        if not cls._fermi_cat:
            cls._fermi_cat = 'downloaded'
        return cls._fermi_cat

print(Fermi3FGLObject._fermi_cat)
f = Fermi3FGLObject()
print(Fermi3FGLObject._fermi_cat)

## -- End pasted text --
None
downloaded
```
In the real code you should call `cls._fermi_cat = fetch_fermi_catalog('3FGL')` instead of `cls._fermi_cat = downloaded'`.